osquery/tables/system/windows/cpu_info.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/sql.h>
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genCpuInfo(QueryContext& context) {
+  Row r;
+  QueryData results;
+
+  WmiRequest wmiSystemReq("SELECT * FROM Win32_Processor");
+  std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
+  if (wmiResults.empty()) {
+    LOG(WARNING) << "Error retreiving information from WMI.";
+    return results;
+  }
+  for (const auto& data : wmiResults) {
```

Comment:
Each of the gets might fail and we need to know that and log it.

Reply:
Would you mean to log an ERROR for each value that might fail if it doesn't return anything?

Reply:
For each `Get<Data Type` value, they typically return a default, but yeah you could at the very least surface the message you get back from them, however this has not been common practice throughout the rest of the code base.

Reply:
You don't necessarily need to log, actually I wouldn't log anything, but if the getters return an error you're going to be using an invalid value, so you need to handle that. Check for instance GetLong on `wmi.cpp`, if an error is returned number is going to be untouched, so your code would just repeat for the new column whatever was used for the column before.

Reply:
Been swamped lately :( Plan to get to this tomorrow or this week If I can...

Reply:
Ok, added some ternaries to handle those errors. I wasn't sure what to do with the value in an error case, so I set it to -1 to match what some of the other tables have. I can change that if needed.